### ISSUE TITLE
Get-AzSqlDatabaseImportExportStatus doesn't return correctly for import with error

### ISSUE THREAD

**Issue by gkingston** [Azure/azure-powershell#10015]
## Description

Get-AzSqlDatabaseImportExportStatus doesn't return a AzureSqlDatabaseImportExportStatusModel when the import had an error.  It give a "Bad Request" and spits the importer error right to the console

## Steps to reproduce
Run a command to get the status with a link to a failed import

```
$importStatus = Get-AzSqlDatabaseImportExportStatus -OperationStatusLink $statusLink
```
Instead of getting an object returned, you'll get a message like the following:
```
Get-AzSqlDatabaseImportExportStatus : BadRequest: The ImportExport operation wit
h Request Id 'xxxxxxxxxxx' failed due to 'Error encount
ered during the service operation.
        Could not import package.
```

## Environment data

```powershell
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.3
PSEdition                      Core
GitCommitId                    v6.0.3
OS                             Microsoft Windows 6.3.9600
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

## Module versions

```powershell
PS C:\> Get-Module -ListAvailable


    Directory: C:\Program Files\PowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     1.6.0      Az
Script     1.4.0      Az.Accounts                         {Disable-AzDataCol...
Script     1.0.1      Az.Aks                              {Get-AzAks, New-Az...
Script     1.0.2      Az.AnalysisServices                 {Resume-AzAnalysis...
Script     1.0.0      Az.ApiManagement                    {Add-AzApiManageme...
Script     1.0.0      Az.ApplicationInsights              {Get-AzApplication...
Script     1.2.0      Az.Automation                       {Get-AzAutomationH...
Script     1.0.0      Az.Batch                            {Remove-AzBatchAcc...
Script     1.0.0      Az.Billing                          {Get-AzBillingInvo...
Script     1.1.0      Az.Cdn                              {Get-AzCdnProfile,...
Script     1.0.1      Az.CognitiveServices                {Get-AzCognitiveSe...
Script     1.6.0      Az.Compute                          {Remove-AzAvailabi...
Script     1.0.0      Az.ContainerInstance                {New-AzContainerGr...
Script     1.0.1      Az.ContainerRegistry                {New-AzContainerRe...
Script     1.0.2      Az.DataFactory                      {Set-AzDataFactory...
Script     1.0.0      Az.DataLakeAnalytics                {Get-AzDataLakeAna...
Script     1.1.0      Az.DataLakeStore                    {Get-AzDataLakeSto...
Script     1.0.0      Az.DevTestLabs                      {Get-AzDtlAllowedV...
Script     1.0.0      Az.Dns                              {Get-AzDnsRecordSe...
Script     1.1.0      Az.EventGrid                        {New-AzEventGridTo...
Script     1.0.1      Az.EventHub                         {New-AzEventHubNam...
Script     1.0.0      Az.HDInsight                        {Get-AzHDInsightJo...
Script     1.0.2      Az.IotHub                           {Add-AzIotHubKey, ...
Script     1.1.0      Az.KeyVault                         {Add-AzKeyVaultCer...
Script     1.2.1      Az.LogicApp                         {Get-AzIntegration...
Script     1.0.0      Az.MachineLearning                  {Move-AzMlCommitme...
Script     1.0.0      Az.MarketplaceOrdering              {Get-AzMarketplace...
Script     1.0.0      Az.Media                            {Sync-AzMediaServi...
Script     1.0.1      Az.Monitor                          {Get-AzMetricDefin...
Script     1.6.0      Az.Network                          {Add-AzApplication...
Script     1.0.0      Az.NotificationHubs                 {Get-AzNotificatio...
Script     1.1.0      Az.OperationalInsights              {New-AzOperational...
Script     1.0.0      Az.PolicyInsights                   {Get-AzPolicyEvent...
Script     1.0.0      Az.PowerBIEmbedded                  {Remove-AzPowerBIW...
Script     1.2.0      Az.RecoveryServices                 {Get-AzRecoverySer...
Script     1.0.0      Az.RedisCache                       {Remove-AzRedisCac...
Script     1.0.0      Az.Relay                            {New-AzRelayNamesp...
Script     1.2.1      Az.Resources                        {Get-AzProviderOpe...
Script     1.0.0      Az.ServiceBus                       {New-AzServiceBusN...
Script     1.0.1      Az.ServiceFabric                    {Add-AzServiceFabr...
Script     1.0.2      Az.SignalR                          {New-AzSignalR, Ge...
Script     1.7.0      Az.Sql                              {Get-AzSqlDatabase...
Script     1.1.0      Az.Storage                          {Get-AzStorageAcco...
Script     1.0.0      Az.StreamAnalytics                  {Get-AzStreamAnaly...
Script     1.0.1      Az.TrafficManager                   {Add-AzTrafficMana...
Script     1.1.2      Az.Websites                         {Get-AzAppServiceP...

```

 ## Expected Output

When an import fails at Azure, a AzureSqlDatabaseImportExportStatusModel should still be returned.  Status should be "Failed" like it shows in the portal and Error Message should be filled out with the actual error

**Comment by aamalvea** [comment]
@maboja-msft Can you take a look at this?

**Comment by dalshamaa** [comment]
Ack

**Comment by sredman** [comment]
Hi @gkingston,
Thank you for taking the time to report this. I agree that this is something which should be fixed. I have put it in our backlog though at the moment our team is tied up with other things. I will keep this issue open for discussion in case someone else wants to look in to it.

**Comment by daniel-simpson** [comment]
Consistently having the same issue, where a database fails to export and receiving a Bad Request error message.

This make it much more difficult to find the root cause of the issue as it looks like a failure to communicate between Azure services rather than a DB backup issue

**Comment by jakeWatsonOakland** [comment]
Looks like the cause is the API itself not PowerShell: 

![image](https://user-images.githubusercontent.com/48357837/119355552-506adc80-bc9d-11eb-9c24-c22d5983eade.png)

**Comment by Valkozaur** [comment]
@daniel-simpson  Hello Daniel! 

I'm experiencing the same kind of issue right now, do you remember how did you solved it. When I run the export from a powershell script it exports for about 5m and than just returns 

Get-AzSqlDatabaseImportExportStatus : Response status code does not indicate success: 400 (Bad Request).

No information at all, this is pretty irritating. Please people, if you have any suggestions it would be great :)

**Comment by daniel-simpson** [comment]
Sorry @Valkozaur, I've changed jobs and don't have access to the fix I made earlier in the year.

From memory I rewrote our export tool from Azure Powershell to using [sqlpackage](https://docs.microsoft.com/en-us/sql/tools/sqlpackage/sqlpackage?view=sql-server-ver15) on an Azure DevOps agent, which means I got the same functionality without the weird error state.

Very annoying that the tooling/API doesn't give a better error message!
